### PR TITLE
Feat/variable hash functions testing

### PR DESCRIPTION
**Pull request by laudiacay**
This adds tests, audits errors, and checks over docs to merge this. Also bumps the version.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module github.com/multiformats/go-multihash
+module github.com/laudiacay/go-multihash
```
Comment (aschmahmann) [inline go.mod:1]: You shouldn't be renaming the module, even for testing this is going to cause you a boatload of problems. You can use go mod replace directive to help you with imports though (e.g. add the line `replace github.com/multiformats/go-multihash => github.com/laudiacay/go-multihash <commit hash>`)

Comment (laudiacay) [reply go.mod:1]: it won't let me do that and import without problems into kubo? It gets mad that the module name doesn't match the path. It works ok and lets me run `go vet` when I change the name. I'm sorry about all the back and forth, I just don't know what I'm doing with go imports.

Comment (Jorropo) [reply go.mod]: @laudiacay `replace` check against the original name of the module, not the replaced one.

So if I do
```
replace github.com/multiformats/go-multihash => github.com/laudiacay/go-multihash version
```
`github.com/laudiacay/go-multihash`'s module name need to be `github.com/multiformats/go-multihash`.

Also this is temporary while the PRs are in flight to see CI pass, whoever is merging this is gonna make new versions, remove the replaces with proper version bumps.

File: multihash.go [hunk 2]
```diff
@@ -27,11 +27,12 @@ var (
...
 // ErrInconsistentLen is returned when a decoded multihash has an inconsistent length
 type ErrInconsistentLen struct {
-	dm *DecodedMultihash
+	dm          *DecodedMultihash
+	lengthFound int
 }
 
 func (e ErrInconsistentLen) Error() string {
-	return fmt.Sprintf("multihash length inconsistent: expected %d, got %d", e.dm.Length, len(e.dm.Digest))
+	return fmt.Sprintf("multihash length inconsistent: expected len(hdig)=%d, got rlen=%d", e.dm.Length, e.lengthFound)
```
Comment (Jorropo) [inline multihash.go:35]: Go's convention is to not include specific implementation details like variable names in error messages.
```suggestion
	return fmt.Sprintf("multihash length inconsistent: expected %d; got %d", e.dm.Length, e.lengthFound)
```

File: core/registry.go [hunk 3]
```diff
@@ -10,15 +10,16 @@ import (
...
+// The function may error (i.e., return (nil, false)) to signify that the hasher can't return that many bytes.
 //
 // Multihash indicator numbers are reserved and described in
 // https://github.com/multiformats/multicodec/blob/master/table.csv .
 // The keys used in this map must match those reservations.
 //
 // Hashers which are available in the golang stdlib will be registered automatically.
 // Others can be added using the Register function.
-var registry = make(map[uint64]func() hash.Hash)
+var registry = make(map[uint64]func(int) (hash.Hash, bool))
```
Comment (Jorropo) [inline core/registry.go:22]:
```suggestion
var registry = make(map[uint64]func(int) (h hash.Hash, ok bool))
```

File: register/blake3/multihash_blake3.go [hunk 4]
```diff
@@ -18,6 +18,6 @@ import (
 )
 
 func init() {
-	multihash.Register(multihash.BLAKE3, func() hash.Hash { h := blake3.New(32, nil); return h })
+	multihash.Register(multihash.BLAKE3, func() hash.Hash { h := blake3.New(128, nil); return h })
```
Comment (Jorropo) [inline register/blake3/multihash_blake3.go:21]: Shouldn't it call `RegisterVariableSize` ?
```suggestion
	multihash.RegisterVariableSize(multihash.BLAKE3, func(size int) (hash.Hash, bool) {
	  if size == -1 {
	    size = 32
	  } else if size > 128 || size <= 0 {
	    return nil, false
	  }
	  h := blake3.New(size, nil)
	  return h, true
	})
```
I would like for the magic numbers to be unmagified if possible, if it's untidy to do so, just skip it.

**Review by Jorropo — CHANGES_REQUESTED** [review]
Small details left to solve. :slightly_smiling_face:

**Comment by Jorropo** [comment]
PLS also rebase on master.

**Review by Jorropo — APPROVED** [review]
LGTM (I'll do fixup)

**Comment by Jorropo** [comment]
replaced with #161

**Pull request closed without merging**